File: spring-data-requery-java/src/main/java/com/coupang/springframework/data/requery/provider/ProxyIdAccessor.java
package com.coupang.springframework.data.requery.provider;

import org.jetbrains.annotations.Nullable;

/**
 * com.coupang.springframework.data.requery.provider.ProxyIdAccessor
 *
 * @author debop
 * @since 18. 6. 7
 */
public interface ProxyIdAccessor {

    boolean shouldUseAccessorFor(Object entity);

    @Nullable
    Object getIdentifierFrom(Object entity);
}

File: spring-data-requery-kotlin/src/main/kotlin/org/springframework/data/requery/kotlin/provider/ProxyIdAccessor.kt
package org.springframework.data.requery.kotlin.provider

/**
 * org.springframework.data.requery.provider.ProxyIdAccessor
 *
 * @author debop
 */
interface ProxyIdAccessor {

    fun shouldUseAccessorFor(entity: Any?): Boolean

    fun getIdentifierFrom(entity: Any?): Any?
}